From e37449b22ae91f0d58f2687b31485b0e4b71ff42 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Thu, 14 Dec 2023 04:40:50 +0100
Subject: [PATCH 5/6] Revert "HID: amd_sfh: Check that sensors are enabled
 before set/get report"

This reverts commit a6ad322f27f7c6c91a77ca14898d52393ed8297d.

This quite old patch was rejected from mainline. Please see
https://lore.kernel.org/all/89ea9fb7-9026-ccb6-ad88-50e1c28b4474@amd.com/

If you still encounter this issue with the latest firmware for your
machine, please respond to that thread.
---
 drivers/hid/amd-sfh-hid/amd_sfh_client.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/hid/amd-sfh-hid/amd_sfh_client.c b/drivers/hid/amd-sfh-hid/amd_sfh_client.c
index 0f1d677..bdb578e 100644
--- a/drivers/hid/amd-sfh-hid/amd_sfh_client.c
+++ b/drivers/hid/amd-sfh-hid/amd_sfh_client.c
@@ -25,9 +25,6 @@ void amd_sfh_set_report(struct hid_device *hid, int report_id,
 	struct amdtp_cl_data *cli_data = hid_data->cli_data;
 	int i;
 
-	if (!cli_data->is_any_sensor_enabled)
-		return;
-
 	for (i = 0; i < cli_data->num_hid_devices; i++) {
 		if (cli_data->hid_sensor_hubs[i] == hid) {
 			cli_data->cur_hid_dev = i;
@@ -44,9 +41,6 @@ int amd_sfh_get_report(struct hid_device *hid, int report_id, int report_type)
 	struct request_list *req_list = &cli_data->req_list;
 	int i;
 
-	if (!cli_data->is_any_sensor_enabled)
-		return -ENODEV;
-
 	for (i = 0; i < cli_data->num_hid_devices; i++) {
 		if (cli_data->hid_sensor_hubs[i] == hid) {
 			struct request_list *new = kzalloc(sizeof(*new), GFP_KERNEL);
-- 
2.43.0.rc2

